File: src/my_project/easy_problems/from101to150/baseball_game.py
from typing import List 

class Solution:
    def calPoints(self, ops: List[str]) -> int:
        solution: List[int] = list()
        for item in ops:
            if item == "C":
                solution.pop()
                continue
            elif item == "D":
                solution.append(int(solution[-1])*2)
                continue
            elif item == "+":
                solution.append(int(solution[-1]) + int(solution[-2]))
                continue
            else:
                solution.append(int(item))


        return sum(solution )


solution = Solution()


print(solution.calPoints(ops = ["5","2","C","D","+"]))
print(solution.calPoints( ops = ["5","-2","4","C","D","9","+","+"]))

File: src/my_project_typescript/easy_problems_typescript/from1to50/baseballGame.ts
function calPoints(ops: string[]): number {


    let solution: Array<number> = [];
    let count: number = 0

    for (let item of ops){
        if (item === "C"){
            solution.pop()
        }
        else if (item === "D"){
            solution.push(solution[solution.length-1]*2)
        }
        else if (item === "+"){
            solution.push(solution[solution.length-1] + solution[solution.length-2])
        }
        else {
            solution.push(parseInt(item))
        }
    }

    solution.map( (currentElement) => {
        count += currentElement
        return count
    })


    return count
};


console.log(calPoints(["5","2","C","D","+"]))